.github/workflows/ci.yml

```diff
@@ -0,0 +1,80 @@
+on: push
```

Comment:
I would suggest this being

```yaml
on:
  pull_request:
  push:
    branches:
      - master
```

This way the CI is only run for pull requests and when pushing to master.

Reply:
Alright.  Is there a reason not to want the continuous validation for every push?